src/main/java/use_case/getpost/GetPostInteractor.java

```diff
@@ -0,0 +1,43 @@
+package use_case.getpost;
+
+import entity.Post;
+import exception.PostNotFoundException;
+
+/**
+ * The Get Post Interactor.
+ */
+public class GetPostInteractor implements GetPostInputBoundary {
+    private final GetPostInputData getPostInputData;
+    private final GetPostDataAccessInterface postDB;
+    private final GetPostOutputBoundary getPostPresenter;
+
+    public GetPostInteractor(GetPostInputData getPostInputData,
+                             GetPostDataAccessInterface postDB,
+                             GetPostOutputBoundary getPostPresenter) {
+        this.getPostInputData = getPostInputData;
+        this.postDB = postDB;
+        this.getPostPresenter = getPostPresenter;
+    }
+
+    @Override
+    public Post getPost(GetPostInputData postInputData) throws PostNotFoundException {
+        final String entryID = postInputData.getEntryID();
+
+        if (entryID == null) {
+            throw new IllegalArgumentException();
+        }
+
+        try {
+            final Post retrievedPost = postDB.getPostByEntryID(entryID);
+            final GetPostOutputData retrievedPostOutputData = new GetPostOutputData(
+                    retrievedPost.getEntryID(),
+                    retrievedPost.getContent());
+            getPostPresenter.prepareSuccessView(retrievedPostOutputData);
+            return retrievedPost;
+        }
+        catch (PostNotFoundException ex) {
+            getPostPresenter.prepareFailView(ex.getMessage());
+            throw ex;
```

Comment:
I dont think we need to throw the exception, that would break the program instead of giving a fail view

Reply:
Oh I see what you mean, but I'm pretty sure the exception here wouldn't break the program as it is thrown after calling prepareFailView and also I added this for testing (because I need to verify both that the fail view is shown and that the exception is properly propagated in the test scenario)

And the reason I added exception propagation is so that we could also do further handling  (if we decide to) such as displaying a "Can't retreive post" on the homepage for a particular post if it can't be retrieved, etc. 

Lemme know if you still want me to remove it (though i'd have to change the tests [here ](https://github.com/csc207group76/connecthub/pull/9/files#diff-f2489a4749b0ff1da634585d2c368e4335418abfdfa9ba5a9a362cac60cba700R68)too 😅)

Reply:
Personally I would prefer not throwing the exception - we can display custom fail messages in the getPostPresenter, but we could check with the TA tmrw :)

Reply:
Commenting here: TA said we can throw exceptions here but also it's more typical for an application to log them instead. 

I can remove this for now + will remove the related test

Reply:
Removed the exceptions except for a null entryID! Lemme know if this is ok (+ modified the tests)